### PR TITLE
feat: support ffi

### LINKED ISSUE #4: Type error
OS: MacOS Catalina
Deno version: 1.9.2
Argon version: 0.9.2

Hi @fdionisi ! I am trying to use the argon library in an API which I am making, but I am getting this error:

```shell
error: Uncaught (in promise) TypeError: invalid promise id: expected type `rusty_v8::data::Integer`, got `rusty_v8::data::Value`
      let result = Deno.core.dispatch(argon2_hash, args, buf)!;
                             ^
    at Object.hash (https://deno.land/x/argon2@v0.9.2/lib/internal.ts:97:30)
make: *** [run] Error 1
```

You can reproduce the issue with the following code:
```ts
import { hash, Variant } from "https://deno.land/x/argon2@v0.9.2/lib/mod.ts";

const secretString = "some-secret";

const encoder = new TextEncoder();

const secret = encoder.encode(secretString);

const passwordHash = await hash("password", {
  variant: Variant.Argon2id,
  secret,
});

console.log(passwordHash);
```

### PR DESCRIPTION
**Pull request by cake-monotone**
PR summary:

- removed all the codes for `deno plugin` which is never working currently
- implementation for deno FFI API
- bumped up the version of dependencies
- because `lib/dev.ts` was not able to implement with ffi, I made "./test-with-local-builld.sh"

Could you please review the `native` codes more carefully? I'm not yet familiar with Rust perfectly.

fix #4

### CODE REVIEW

**Comment by fdionisi** [comment]
This is awesome! Thank you, @cake-monotone, for your contribution. 🙏
I'll surely take some time to review it during the weekend.

**Comment by fdionisi** [comment]
Hey @cake-monotone, I was wondering if there's a specific reason not to use https://github.com/denoland/deno_bindgen. By using it, we could remove the serialization part. Back in the day, I decided to use `serde` because I wanted to pass an object, which wasn't possible without this trick.

**Comment by cake-monotone** [comment]
I actually used `deno_bindgen` when I first implemented codes for ffi.

After several attempts, I got the conclusion that `deno_bindgen` is not useful yet.

The biggest reason is memory leak:
https://github.com/denoland/deno_bindgen/issues/108
Internally, there is no logic to free the pointer after sending it to deno.

Also, I experienced random failures in the test code. I have no idea why it occurred, but I think it's a pointer and memory problem in deno_bindgen

File: lib/mod.ts [hunk 1]
```diff
@@ -56,7 +48,7 @@ export async function hash(
   password: string,
   options: Partial<HashOptions> = {},
 ) {
-  return plugin.hash(password, options);
+  return await internal.hash(password, options);
```
Comment (fdionisi) [inline lib/mod.ts]: I believe we should be able to remove the `await` here and the `async` in the function signature as well.

File: lib/mod.ts [hunk 2]
```diff
@@ -68,5 +60,5 @@ export async function verify(
   hash: string,
   password: string,
 ) {
-  return plugin.verify(hash, password);
+  return await internal.verify(hash, password);
```
Comment (fdionisi) [inline lib/mod.ts]: Ditto

**Review by fdionisi — APPROVED** [review]
I see. Thank you for the explanation, @cake-monotone.

Your solution is a good first iteration to get FFI in the codebase.

Later I'd like to explore how/if errors propagate from Rust to Deno and potentially avoid deserializing input and serializing outputs.

I have two quick notes before merging, but I think this is ok for enabling the library with newer Deno versions.

Once again, thank you very much for your contribution. 🙏

**Comment by cake-monotone** [comment]
Thank you for your kind review and comments!
I hope this PR helps

**Pull request merged**